homeworks/natalia.kokovina_NataliKokovina/homework_6/task2.js

```diff
@@ -0,0 +1,73 @@
+const horsName = ['Jake', 'Coco', 'Max', 'Rain', 'Star', 'Misty', 'Blaze', 'Lucky', 'Lily', 'Duke'];
+const horsBreed = ['Arabian', 'Quarter', 'Thoroughbred', 'Tennessee', 'Morgan', 'Paint', 'Appaloosa', 'Miniature', 'Warmblood', 'Andalusian'];
+
+class Horse {
+    constructor(name, breed) {
+        this.name = name;
+        this.breed = breed;
+    }
+}
+
+class Racer extends Horse {
+    constructor(name, breed) {
+        super(name, breed);
+        this.distance = 0;
+        this.speed = 0;
+    }
+
+    setSpeed() {
+        this.speed = Math.floor(Math.random() * (15 - 10) + 10);
+    }
+
+    run() {
+        for (let i = 1; i <= 10; i++) {
+            setTimeout(() => {
+                this.setSpeed();
+                this.distance += this.speed;
+            }, 1000 * i);
+        }
+    }
+}
+
+class Race {
+    constructor() {
+        this.horses = [];
+    }
+
+    crearRace() {
+        const allHorses = [];
+        for (let i = 0; i <= horsName.length - 1; i++) {
+            const hors = new Horse(horsName[i], horsBreed[i]);
+            allHorses.push(hors);
```

Comment:
hmmm... didn't get why do you need this, instantiating `Racer` will do what you need as `Racer` extends `Horse`

Reply:
I used this cycle instead manual write const allHorses like
const allHorses = [{ name: "..." , breed: "..."}, { name: "..." , breed: "..."} ... ]
isnt it correrect? or its can write in Racer or something else?

Reply:
Made it easier